File: custom_types/FluffySink.py
import Pothos

class FluffySink(Pothos.Block):
    def __init__(self):
        Pothos.Block.__init__(self)
        self.setupInput(0)

    def work(self):
        inPort = self.input(0)

        #do we have an input message?
        if not inPort.hasMessage(): return

        #extract the data
        data = inPort.popMessage()

        #print the data
        print("FluffySinkPy: fluff=%d"%data.getFluff())
        print("FluffySinkPy: wiggles=%s"%data.wiggles)

File: custom_types/FluffySink.cpp
#include <Pothos/Framework.hpp>
#include "FluffyData.hpp" //use class definition in sink version 1
#include <iostream>

class FluffySink : public Pothos::Block
{
public:
    static Block *make(void)
    {
        return new FluffySink();
    }

    FluffySink(void)
    {
        this->setupInput(0);
    }

    void work(void)
    {
        auto inPort = this->input(0);

        //do we have an input message?
        if (not inPort->hasMessage()) return;

        //extract the data
        const auto msg = inPort->popMessage();
        const auto &data = msg.extract<FluffyData>();

        //print the data
        std::cout << "FluffySink: fluff=" << data.getFluff() << std::endl;
        std::cout << "FluffySink: wiggles=" << data.wiggles << std::endl;
    }
};

static Pothos::BlockRegistry registerFluffySink(
    "/demos/fluffy_sink", &FluffySink::make);
